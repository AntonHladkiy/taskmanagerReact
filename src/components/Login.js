import React, { useState } from 'react';
import {Link, Redirect} from "react-router-dom";
const Login = props => {
    const [user, setUser] = useState(props.initialUser);

    const handleInputChange = event => {
        const { name, value } = event.target
        setUser({ ...user, [name]: value })
    };
    if(props.loggedIn===true){
        return <Redirect to="/"></Redirect>
    }
    return (
        <form className={"form-check"} autocomplete="off">
            <label className= "form-check-label">Email</label>
            <input className= "form-control w-25" type="text" name="email" value={user.email} onChange={handleInputChange} ></input>
            <label className= "form-check-label">Password</label>
            <input className= "form-control w-25" type="text" name="password"   value={user.password} onChange={handleInputChange} ></input>
            <Link to ="/"> <button className="btn btn-warning mt-2 mr-2" onClick={()=> {
                if (!user.email|| !user.password) return;
                props.logIn(user)
            }}>Log In</button></Link>
            <Link to ="/"><button className={"btn btn-danger mt-2"}>Cancel</button></Link>
        </form>
    )
}
export default Login;