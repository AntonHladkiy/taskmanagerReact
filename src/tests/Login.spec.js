import React from 'react';
import { shallow} from 'enzyme';
import Login from '../components/Login';
let initialUser={
    email:'',
    password:''
}
describe('Login', () => {
    let wrapped = shallow(<Login initialUser={initialUser} />);
    it('should render the Login Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
});
describe('Login when already loggenIn', () => {
    let wrapped = shallow(<Login initialUser={initialUser} loggedIn={true}/>);
    it('should render the Login Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
});