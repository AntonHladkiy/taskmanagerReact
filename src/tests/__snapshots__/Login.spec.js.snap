// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the Login Component correctly 1`] = `
<form
  autoComplete="off"
  className="form-check"
>
  <label
    className="form-check-label"
  >
    Email
  </label>
  <input
    className="form-control w-50"
    name="email"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    className="form-check-label"
  >
    Password
  </label>
  <input
    className="form-control w-50"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <Link
    to="/"
  >
     
    <button
      className="btn btn-success mt-2 mr-2 w-25"
      onClick={[Function]}
    >
      Log In
    </button>
  </Link>
  <Link
    to="/"
  >
    <button
      className="btn btn-danger mt-2 w-25"
    >
      Cancel
    </button>
  </Link>
</form>
`;

exports[`Login when already loggenIn should render the Login Component correctly 1`] = `
<Redirect
  to="/"
/>
`;
